jwst/pipeline/calwebb_spec3.py

```diff
@@ -317,3 +331,26 @@ def _create_nrsfs_slit_name(self, source_models):
         slit_name = "-".join(slit_names)  # append slit names using a dash separator
 
         return slit_name
+
+
+# #########
+# Utilities
+# #########
+def invariant_filename(save_model_func):
```

Comment:
This utility appears to be generic, not limited only to calwebb_spec3.  I would suggest it be moved to a more generic location.  Maybe @hbushouse has a suggestion of where that would be, but perhaps something like making a new pipeline/utils.py file and import from there?